custom/icds_reports/utils/aggregation_helpers/distributed/ls_vhnd_form.py

```diff
@@ -30,14 +30,15 @@ def aggregate_query(self):
                 state_id,
                 location_id as supervisor_id,
                 %(start_date)s::DATE AS month,
-                count(*) as vhnd_observed
+                count(*) FILTER (WHERE submitted_on >= %(start_date)s AND
+                   submitted_on < %(end_date)s) as form_count,
+                count(*) FILTER (WHERE vhnd_date >= %(start_date)s AND vhnd_date < %(end_date)s) as vhnd_observed
                 FROM "{ucr_tablename}"
-                WHERE vhnd_date >= %(start_date)s AND vhnd_date < %(end_date)s
-                AND state_id=%(state_id)s
+                WHERE state_id=%(state_id)s
```

Comment:
This query looks very inefficient. I think it has to do a full table scan always, since different columns filter on different date ranges. Can you run `EXPLAIN` on this, on the real table to see what it looks like. This could be incredibly expensive as the table grows.

Reply:
@calellowitz 
This is what explain gives
```
icds_ucr=> EXPLAIN CREATE TEMPORARY TABLE "temp_testing_delete" AS (
icds_ucr(>     SELECT
icds_ucr(>         state_id,
icds_ucr(>         location_id as supervisor_id,
icds_ucr(>         '2019-11-01'::DATE AS month,
icds_ucr(>         count(*) FILTER (WHERE submitted_on >= '2019-11-01' AND
icds_ucr(>         submitted_on < '2019-11-30') as form_count,
icds_ucr(>         count(*) FILTER (WHERE vhnd_date >= '2019-11-01' AND vhnd_date < '2019-11-30') as vhnd_observed
icds_ucr(>         FROM "ucr_icds-test_static-ls_vhnd_form_f6161093"
icds_ucr(>         GROUP BY state_id,location_id
icds_ucr(> );
                                                                                 QUERY PLAN                                                                                 
----------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 HashAggregate  (cost=0.00..0.00 rows=0 width=0)
   Group Key: remote_scan.state_id, remote_scan.supervisor_id
   ->  Custom Scan (Citus Real-Time)  (cost=0.00..0.00 rows=0 width=0)
         Task Count: 64
         Tasks Shown: One of 64
         ->  Task
               Node: host=100.71.184.232 port=5432 dbname=icds_ucr
               ->  HashAggregate  (cost=26.80..28.80 rows=200 width=84)
                     Group Key: state_id, location_id
                     ->  Seq Scan on "ucr_icds-test_static-ls_vhnd_form_f6161093_119866" "ucr_icds-test_static-ls_vhnd_form_f6161093"  (cost=0.00..15.60 rows=560 width=76)
(10 rows)
```

Reply:
@calellowitz  THis is what Explain gives
```
 icds_ucr=> EXPLAIN CREATE TEMPORARY TABLE "temp_testing_delete" AS (
icds_ucr(>     SELECT
icds_ucr(>         state_id,
icds_ucr(>         location_id as supervisor_id,
icds_ucr(>         '2019-11-01'::DATE AS month,
icds_ucr(>         count(*) FILTER (WHERE submitted_on >= '2019-11-01' AND
icds_ucr(>         submitted_on < '2019-11-30') as form_count,
icds_ucr(>         count(*) FILTER (WHERE vhnd_date >= '2019-11-01' AND vhnd_date < '2019-11-30') as vhnd_observed
icds_ucr(>         FROM "ucr_icds-cas_static-ls_vhnd_form_f2b97e26"
icds_ucr(>         GROUP BY state_id,location_id
icds_ucr(> );
                                                                                QUERY PLAN                                                                                
--------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 HashAggregate  (cost=0.00..0.00 rows=0 width=0)
   Group Key: remote_scan.state_id, remote_scan.supervisor_id
   ->  Custom Scan (Citus Real-Time)  (cost=0.00..0.00 rows=0 width=0)
         Task Count: 64
         Tasks Shown: One of 64
         ->  Task
               Node: host=100.71.184.232 port=5432 dbname=icds_ucr
               ->  HashAggregate  (cost=44.85..45.75 rows=90 width=86)
                     Group Key: state_id, location_id
                     ->  Seq Scan on "ucr_icds-cas_static-ls_vhnd_form_f2b97e26_102970" "ucr_icds-cas_static-ls_vhnd_form_f2b97e26"  (cost=0.00..26.95 rows=895 width=78)
(10 rows)
```
this ucr has 44K records at the moment

Reply:
Talked to cal a bit on slack about this. This is already doing a sequential scan anyways because there are no indexes, so this should cause no worse performance than existing code.

I think that this is likely small enough to be fine for now, but may need to be addressed in the future depending on how fast this table grows. Some changes that could happen:

1) index state_id (note that that exists in the query in code, but not your explain)
2) index vhnd_date and submitted_on and add where clause `submitted_on >= date OR vhnd-date >= date` (this wouldn't be super ideal as it would only be able to use one, or need to bitmap scan both indexes.)